speech/norm/__init__.py

```diff
@@ -231,42 +182,244 @@ def _email_handler(email: str) -> str:
+        return numbers_to_text(txt, **kwargs)
+
+    @classmethod
+    def float(cls, txt: str, **kwargs: str):
+        """Voicify a float."""
+        return float_to_text(txt, **kwargs)
+
+    @classmethod
+    def floats(cls, txt: str, **kwargs: str):
+        """Voicify text containing multiple floats."""
+        return floats_to_text(txt, **kwargs)
+
+    @classmethod
+    def ordinal(cls, txt: str, **kwargs: str):
+        """Voicify an ordinal."""
+        return number_to_ordinal(txt, **kwargs)
+
+    @classmethod
+    def ordinals(cls, txt: str, **kwargs: str):
+        """Voicify text containing multiple ordinals."""
+        return numbers_to_ordinal(txt, **kwargs)
+
+    @classmethod
+    def digits(cls, txt: str):
+        """Spell out digits."""
+        return digits_to_text(txt)
+
+    @classmethod
+    def phone(cls, txt: str):
+        """Spell out digits."""
+        return cls.digits(txt)
+
+    @classmethod
+    def time(cls, txt: str):
+        """
+        Voicifies time of day, specified as 'HH:MM'.
+        E.g.
+            "11:34" -> "ellefu þrjátíu og fjögur",
+            "00:30" -> "tólf þrjátíu um nótt"
+        Note: doesn't check for incorrect data, caller should handle.
+        """
+        h, m = [int(i) for i in txt.split(":")]
+        suffix: Optional[str] = None
+        # Some times
+        if h == 0:
+            # Call 00:00 "tólf á miðnætti"
+            # and 00:xx "tólf ... um nótt"
+            h = 12
+            suffix = "á miðnætti" if m == 0 else "um nótt"
+        elif 0 < h <= 5:
+            suffix = "um nótt"
+        elif h == 12 and m == 0:
+            suffix = "um hádegi"
+        t = [
+            number_to_text(h, case="nf", gender="hk"),
+        ]
+        if m > 0:
+            if m < 10:
+                # e.g. "þrettán núll fjögur"
+                t.append("núll")
+            t.append(number_to_text(m, case="nf", gender="hk"))
+        if suffix:
+            t.append(suffix)
+        return " ".join(t)
+
+    @classmethod
+    def date(cls, txt: str, case: str = "nf"):
+        """
+        Voicifies dates specified in either
+            'YYYY-MM-DD' (ISO 8601 format),
+            'DD/MM/YYYY' or
+            'DD. month[ YYYY]'
+        Note: doesn't check for incorrect numbers,
+            as that should be handled by caller.
+        """
+        # Get first fullmatch from date regexes
+        m = next(
+            filter(
+                lambda x: x is not None,
+                (r.fullmatch(txt) for r in _DATE_REGEXES),
+            ),
+            None,  # Default if no matches are found
+        )
+        assert m is not None, f"Incorrect date format specified for date handler: {txt}"
+
+        # Handle 'DD/MM/YYYY' or 'MM. jan/feb/... [year]' match
+        gd = m.groupdict()
+        day = number_to_ordinal(gd["day"], gender="kk", case=case, number="et")
+        # Month names don't change in different declensions
+        month = (
```

Comment:
A little bit inefficient to have three lookups of gd["month"] here; a temp variable might be in order

Reply:
Fixed